### PR TITLE
Feat: Project reward form refactor

### PR DESCRIPTION
**Pull request by Agill-Sheron**

### CODE REVIEW

File: src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  useProjectRewardCurrencyUpdateMutation,
+  useProjectRewardQuery,
+} from '@/types'
+import { useNotification } from '@/utils'
+
+type FormValues = Omit<ProjectRewardFragment, 'id' | 'sold'>
+
+const MAX_REWARD_IMAGES = 5
+
+const REWARD_CATEGORIES_QUERY = gql`
```
Comment (sajald77) [inline src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts:30]: let's define queries in the graphql folder, it generates types and stays consistent with the rest.

File: src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+
+  const {
+    isOpen: isCurrencyChangeModalOpen,
+    onClose: closeCurrencyChangeModal,
+    onOpen: openCurrencyChangeModal,
+  } = useModal()
+
+  const { getUSDAmount, getSatoshisFromUSDCents } = useBTCConverter()
+
+  console.log('reward', data)
```
Comment (sajald77) [inline src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts:93]: console log.

File: src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    isAddon: yup.boolean(),
+    isHidden: yup.boolean(),
+    category: yup.string().nullable(),
+    preOrder: yup.boolean(),
+    rewardCurrency: yup.string(),
+    estimatedAvailabilityDate: yup.date().nullable(),
+    estimatedDeliveryInWeeks: yup.number().nullable().min(0, 'Delivery time must be greater than or equal to 0'),
+    confirmationMessage: yup.string().max(500, 'Confirmation message must be at most 500 characters long'),
+    privateCommentPrompts: yup.array().of(yup.string()),
+  })
```
Comment (sajald77) [inline src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts:54]: don't see a use of this `solidAmount`  inside this function

Comment (Agill-Sheron) [reply src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts:54]: Yes you're right, I tried to use it to make a proper validation for the `maxClaimable` condition that cannot be lower than the `soldAmount` but it didn't work as I expected - I insist in doing in yup to keep the whole validation consistent. 
Good catch!

**Review by sajald77 — APPROVED** [review]

**Pull request merged**